pkg/statistics/handle/autoanalyze/priorityqueue/calculator.go

```diff
@@ -14,23 +14,57 @@
 
 package priorityqueue
 
-// WeightCalculator is an interface for calculating weights of analysis jobs.
-type WeightCalculator interface {
-	CalculateWeight(job *TableAnalysisJob) float64
-}
+import "math"
+
+const (
+	// EventNone represents no special event.
+	eventNone = 0.0
+	// EventNewIndex represents a special event for newly added indexes.
+	eventNewIndex = 2.0
+)
+
+// TODO: make these configurable.
+const (
+	changeRatioWeight = 0.6
+	sizeWeight        = 0.1
+	analysisInterval  = 0.3
+)
 
 // PriorityCalculator implements the WeightCalculator interface.
-type PriorityCalculator struct {
-	threshold float64
-}
+type PriorityCalculator struct{}
 
-// NewPriorityCalculator creates a new PriorityCalculator with the given threshold.
-func NewPriorityCalculator(threshold float64) *PriorityCalculator {
-	return &PriorityCalculator{threshold: threshold}
+// NewPriorityCalculator creates a new PriorityCalculator.
+//
+// For more information, please visit:
+// https://github.com/pingcap/tidb/blob/master/docs/design/2023-11-29-priority-queue-for-auto-analyze.md
+func NewPriorityCalculator() *PriorityCalculator {
+	return &PriorityCalculator{}
 }
 
 // CalculateWeight calculates the weight based on the given rules.
-func (*PriorityCalculator) CalculateWeight(_ *TableAnalysisJob) float64 {
-	// TODO: implement the weight calculation
-	return 1
+// - Table Change Ratio (Change Ratio): Accounts for 60%
+// - Table Size (Size): Accounts for 10%
+// - Analysis Interval (Analysis Interval): Accounts for 30%
```

Comment:
Personally, I think this is a bit confusing.
At first sight, I thought the Change Ratio would be between [0, 0.6], the Size will be between [0, 0.1] and the Analysis interval will be between [0, 0.3]. But actually, it's not.